Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,284 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ForgeSchemaValidator.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// <summary>
+//     The ForgeSchemaValidator class implements the ITreeSchemaValidator interface.
+// </summary>
+//-----------------------------------------------------------------------
+namespace Microsoft.Forge.TreeWalker
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading.Tasks;
+    using Microsoft.Forge.DataContracts;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using Newtonsoft.Json.Schema;
+    /// <summary>
+    /// The ForgeSchemaValidator class implements the validation method that tests input schemas with custom rules from input.
+    /// </summary>
+    public static class ForgeSchemaValidator
+    {
+        /// <summary>
+        /// The GetLinkedJSchema method that creates the linked rules in JSchema.
+        /// </summary>
+        /// <param name="childRules">The rules to be included in the parent rules</param>
+        /// <param name="parentRules">The parent rules to absorb the child rules</param>
+        /// <param name="referenceUri">The address of childRules</param>
+        /// <returns>The result of schema combination. ErrorMessage would be set if it throw exceptions</returns>
+        public static JSchema GetLinkedJSchemaRules(string childRules, string parentRules, string referenceUri, out string errorMessage)
+        {
+            try
+            {
+                JSchemaPreloadedResolver resolver = new JSchemaPreloadedResolver();
+                resolver.Add(new Uri(referenceUri), childRules);
+                errorMessage = "";
+                return JSchema.Parse(parentRules, resolver);
+            }
+            catch(Exception e)
+            {
+                errorMessage = e.Message;
+                return null;
+            }
```

Comment:
We can remove the try/catch and the "out string errorMessage." If the JSchema.Parse fails, then we want to throw an exception so the end-user can know something went wrong and handle that.